OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs

```diff
@@ -204,13 +204,13 @@ public DefaultSpriteSequence(ModData modData, TileSet tileSet, SpriteCache cache
 							"{0}: Sequence {1}.{2}: Length must be <= Frames.Length"
 							.F(info.Nodes[0].Location, sequence, animation));
 
-					if (Start < 0 || Start + (Facings - 1) * Stride + Length > frameCount)
+					if (Frames == null && (Start < 0 || Start + (Facings - 1) * Stride + Length > frameCount))
```

Comment:
We still need to check this if `Frames != null`!

Simple testcase: 
```
	stand:
		Frames: 0, 1, 2, 3
		Facings: 8
```
on an infantry - this will throw an unhelpful `IndexOutOfRangeException`.

Reply:
That isn't trivial to fix. I tried:

```c-sharp
					if (Frames != null) 
					{
						var distinctFrames = Frames.Distinct().Count();
						if (Start < 0 || Start + (Facings - 1) * Stride + distinctFrames > frameCount)
						throw new InvalidOperationException(
							"{5}: Sequence {0}.{1} uses frames [{2}..{3}], but only 0..{4} actually exist"
							.F(sequence, animation, Start, Start + (Facings - 1) * Stride + distinctFrames - 1, frameCount - 1,
								info.Nodes[0].Location));
					}
```

but it didn't catch your case.

Reply:
```csharp
var end = Start + (Facings - 1) * Stride + Length - 1;
if (Frames != null)
{
	foreach (var f in Frames)
		if (f < 0 || f >= frameCount)
			throw new InvalidOperationException(
				"{5}: Sequence {0}.{1} defines a Frames override that references frame {4}, but only [{2}..{3}] actually exist"
					.F(sequence, animation, Start, end, f, info.Nodes[0].Location));

	if (Start < 0 || end >= Frames.Length)
		throw new InvalidOperationException(
			"{5}: Sequence {0}.{1} uses indices [{2}..{3}] of the Frames list, but only {4} frames are defined"
				.F(sequence, animation, Start, end, Frames.Length, info.Nodes[0].Location));
}
else if (Start < 0 || end >= frameCount)
	throw new InvalidOperationException(
		"{5}: Sequence {0}.{1} uses frames [{2}..{3}], but only 0..{4} actually exist"
			.F(sequence, animation, Start, end, frameCount - 1, info.Nodes[0].Location));
```